polkadot/transaction-pool/src/lib.rs

```diff
@@ -251,8 +329,9 @@ impl TransactionPool {
 		}
 	}
 
+	// TODO: remove. This is pointless - just use `submit()` directly.
```

Comment:
We have 2 places were we submit `runtime::UncheckedExtrinsic` directly, so we don't have to go through serialization:
```
polkadot/service/src/lib.rs
131:			match self.pool.import_unchecked_extrinsic(best_block_id, uxt) {

polkadot/consensus/src/lib.rs
552:			self.transaction_pool.import_unchecked_extrinsic(uxt)
```

Reply:
the implementation shows it's just a shim for the underlying `pool`'s api:

```
self.inner.submit(vec![uxt]).map(|mut v| v.swap_remove(0))
```

if it's really just meant to be a convenience method for submitting a single transaction then fair enough, but in that case it should be named `submit_typed` or some such. "import" suggests it uses different logic to "submit" which (aside from the largely superficial typing) it doesn't.